streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -118,8 +118,19 @@ private void putIfAbsent(final Properties props, final String propertyKey, final
      * Stop the Kafka cluster.
      */
     private void stop() {
+        if (brokers.length > 1) {
+            // delete the topics first to avoid cascading leader elections while shutting down the brokers
+            try {
+                deleteAllTopicsAndWait(60_000L);
```

Comment:
Do we need to wait to avoid the leader election?

Reply:
hmm, good question. I don't know!

Reply:
Nope! Thanks for the heads-up.

Reply:
The updated code here creates just one request to delete all topics, and also just waits for the delete request to be acked, not for the delete to be fully synced to the whole cluster.